src/Avalonia.Controls/Panel.cs

```diff
@@ -59,6 +69,15 @@ public Panel()
             set { SetValue(BackgroundProperty, value); }
         }
 
+        /// <summary>
+        /// Gets whether the <see cref="Panel"/> hosts the items created by an <see cref="ItemsPresenter"/>.
+        /// </summary>
+        public bool IsItemsHost
```

Comment:
The setter can be made `internal`. My reasoning was that some custom third party control might need to enable the "no logical children" behavior on existing panels it might use in its template, without using the `ItemsControl`/`ItemsPresenter` features.

I completely understand wanting to reduce the public API surface. At the same time I think making behavioral flags internal can severely limit third parties. It's always hard to balance! Let me know what you prefer here.

Binding seems very unlikely. Since it's a public property, a style or behavior might want to to react to this property changing (and it would be inconsistent with other properties otherwise). If the setter becomes internal then it won't ever change after being set once, so the direct property can be removed.

Reply:
Yeah, I've learned through bitter experience that it's easier to add features than take them away, so I think it would be better to make this a non-`AvaloniaProperty` and read-only until requirements prove otherwise.

Reply:
Understood!
I made the requested change and rebased the branch onto master.